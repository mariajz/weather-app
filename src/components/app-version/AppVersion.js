import React, { useEffect, useState } from 'react';
import { Text } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { ContentWrapper } from './AppVersion.style';

const AppVersion = () => {
    const [appVersion, setAppVersion] = useState('0.0');
    const [appBuildNumber, setAppBuildNumber] = useState('0');

    const getVersion = async () => {
        const version = DeviceInfo.getVersion();
        setAppVersion(version);
        const buildNumber = DeviceInfo.getBuildNumber();
        setAppBuildNumber(buildNumber);
    };

    useEffect(() => {
        getVersion();
    });

    return (
        <ContentWrapper>
            <Text testID="version">{`Version ${appVersion}(${appBuildNumber})`}</Text>
        </ContentWrapper>
    );
};

export default AppVersion;
