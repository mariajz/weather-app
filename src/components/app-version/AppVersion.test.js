import React from 'react';
import AppVersion from './AppVersion';
import { render } from '@testing-library/react-native';

jest.mock('react-native-device-info', () => ({
    getVersion: () => '1.0',
    getBuildNumber: () => '2',
}));

describe('AppVersion', () => {
    it('Should render AppVersion', () => {
        const container = render(<AppVersion />);

        const { getByTestId } = container;

        expect(container).toMatchSnapshot();
        expect(getByTestId('version').props.children).toBe('Version 1.0(2)');
    });
});
