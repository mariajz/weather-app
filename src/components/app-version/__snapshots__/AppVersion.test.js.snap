// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppVersion Should render AppVersion 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <Text
    testID="version"
  >
    Version 1.0(2)
  </Text>
</View>
`;
